hs_composite_resource/models.py

```diff
@@ -83,10 +86,7 @@ def get_folder_aggregation_object(self, dir_path):
         """
 
         aggregation_path = dir_path[len(self.file_path) + 1:]
-        try:
-            return self.get_aggregation_by_name(aggregation_path)
-        except ObjectDoesNotExist:
-            return None
+        return self.filesetlogicalfile_set.filter(folder=aggregation_path).first()
```

Comment:
This is a large speedup too

Reply:
This part of code is executed more often as the user navigate through folder structure at the UI - so we should see better UI response.